nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CaptureChangeMySQL.java

```diff
@@ -0,0 +1,928 @@
+        pds.add(HOSTS);
+        pds.add(DRIVER_NAME);
+        pds.add(DRIVER_LOCATION);
+        pds.add(USERNAME);
+        pds.add(PASSWORD);
+        pds.add(DATABASE_NAME_PATTERN);
+        pds.add(TABLE_NAME_PATTERN);
+        pds.add(CONNECT_TIMEOUT);
+        pds.add(DIST_CACHE_CLIENT);
+        pds.add(RETRIEVE_ALL_RECORDS);
+        pds.add(INIT_SEQUENCE_ID);
+        pds.add(INIT_BINLOG_FILENAME);
+        pds.add(INIT_BINLOG_POSITION);
+        propDescriptors = Collections.unmodifiableList(pds);
+    }
+
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propDescriptors;
+    }
+
+    @Override
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        if (INIT_BINLOG_FILENAME.getName().equals(propertyDescriptorName)) {
+            return INIT_BINLOG_FILENAME;
+        }
+        if (INIT_BINLOG_POSITION.getName().equals(propertyDescriptorName)) {
+            return INIT_BINLOG_POSITION;
+        }
+        return null;
+    }
+
+    @OnScheduled
+    public void setup(ProcessContext context) {
+
+        final ComponentLog logger = getLogger();
+
+        final StateManager stateManager = context.getStateManager();
+        final StateMap stateMap;
+
+        try {
+            stateMap = stateManager.getState(Scope.LOCAL);
+        } catch (final IOException ioe) {
+            logger.error("Failed to retrieve observed maximum values from the State Manager. Will not attempt "
+                    + "connection until this is accomplished.", ioe);
+            context.yield();
+            return;
+        }
+
+        Map<String, String> statePropertiesMap = new HashMap<>(stateMap.toMap());
+
+        // Pre-store initial sequence ID if none has been set in the state map
+        final PropertyValue initSequenceId = context.getProperty(INIT_SEQUENCE_ID);
+        statePropertiesMap.computeIfAbsent(EventWriter.SEQUENCE_ID_KEY, k -> initSequenceId.isSet() ? initSequenceId.evaluateAttributeExpressions().getValue() : "0");
+
+
+        boolean getAllRecords = context.getProperty(RETRIEVE_ALL_RECORDS).asBoolean();
+
+        // Set current binlog filename to whatever is in State, falling back to the Retrieve All Records then Initial Binlog Filename, if no State variable is present
+        currentBinlogFile = stateMap.get(BinlogEventInfo.BINLOG_FILENAME_KEY);
+        if (StringUtils.isEmpty(currentBinlogFile)) {
+            if (!getAllRecords && context.getProperty(INIT_BINLOG_FILENAME).isSet()) {
+                currentBinlogFile = context.getProperty(INIT_BINLOG_FILENAME).evaluateAttributeExpressions().getValue();
+            } else {
+                // If we're starting from the beginning of all binlogs, the binlog filename must be the empty string (not null)
+                currentBinlogFile = "";
```

Comment:
That's what the Retrieve All Records property is for. If you clear state, then if you set Retrieve All Records to true it starts at the beginning, if you set it to false it starts at the end. If you have state variables for binlog file/position, you start at that location. I will test those paths to make sure they work as advertised, and will double-check the doc to make sure its usage is clear.